### ISSUE TITLE
Applying VESPA to TESS photometry

### ISSUE THREAD

**Issue by campante80** [timothydmorton/VESPA#16]
Hi Tim,

Over the past few days I've been wondering how hard could it be to tune `VESPA` so that one can readily apply it to TESS photometry.

A few things came up:
— A magnitude for the band in which the transit signal is observed must be passed as a constraint (`'TESS'` in such case).
— The call to `TRILEGAL` currently uses `Phot.system='kepler_2mass'`. Found out in the meantime that `TRILEGAL` now also provides TESS magnitudes (simply use `'TESS_2mass'` or else `'TESS_2mass_kepler'`). Checked this directly with Léo Girardi. The `perl` script used in the call to `TRILEGAL` by default sets the simulation's limiting magnitude to be on the 1st mag in the list. This would now be `'TESS'`, so all good there.
— The cadence under `vespa.populations.EclipsePopulation` defaults to Kepler's long cadence (also occurs in other parts of the code). This could simply be tuned, say, if we were to use TESS's 2-min cadence.
— Also, `vespa.transit_basic.ldcoeffs` returns limb-darkening coefficients in the Kepler band. I've generated a new text file (keeping the assumed format) containing these coefficients in the TESS band. It is based on table 5 of [Claret (2018)](https://arxiv.org/abs/1804.10135), i.e., assuming solar metallicity, 2300<T_eff<12000 K, and 2.5<log(g)<6.0. Can make the file available if needed.

Could there be any other obvious aspects that I'm missing?

Cheers, Tiago

**Comment by timothydmorton** [comment]
I'm very glad you're looking into this.  Aside from the details that you've found, there is a big-picture thing that I'm in the middle of working on now, which is being able to to model *specific* false positive scenarios based on *known* blended stars.  This is already a problem for Kepler when there are detected companions, and it will be unavoidable with TESS, where there will undoubtedly be known blended stars in every aperture.  I have been thinking mostly about this big-picture issue, but it's really helpful also for you to look into these more specific things.  Would you be willing to take a stab at making some of what you have identified configurable?  From a user perspective, I think it would be great to be able to just specify the mission (e.g., Kepler/TESS) in the `fpp.ini`, and make sure everything propagates correctly from that (e.g., passing the correct `cadence`; grabbing the right band from the TRILEGAL file, etc.).  If you're able to do even a little bit of work/testing about this, that would be a huge help (I'm trying to keep a number of balls in the air currently, and won't be able to be all in on working on all this for another few months yet).

I think I will also probably need to enable TESS magnitudes within isochrones; I don't think I've done that yet.  @aarondotter, are synthetic TESS magnitudes available for either dartmouth or MIST model grids?

**Comment by timothydmorton** [comment]
@campante80, feel free to let me know if a skype chat at some point would help you get more oriented with the code before trying to do some of this.  As you may have realized, the documentation is far from complete or up-to-date.

**Comment by aarondotter** [comment]
I *think *they are in MIST; can't check right now.  Not in Dartmouth.  The
only problem, in an absolute sense, is that the photometric reference
(Vega) might have some nonzero magnitude but I don't know what that is yet.

Aaron

**Comment by campante80** [comment]
@timothydmorton I'll be performing some additional tests with `vespa` over the course of this week, after which I must divert my attention to other matters, I'm afraid. I'm especially interested in applying `vespa` to low-luminosity RGB candidate hosts and so will be testing the impact of adding spectroscopic/asteroseismic constraints in such scenarios. I plan, however, to revisit `vespa` later this Summer and, time allowing, I'd then consider tuning some of the code.

Regarding the TESS mags within isochrones (or Kepler for that matter): As I understand, stellar model fits don't take these into account (unless an error bar is specified in `fpp.ini`), right?

**Comment by timothydmorton** [comment]
That's correct; the stellar model fits won't take the TESS/Kepler mag into account, but the stellar model grids still need to be able to predict them in order to compute the depth of blended eclipses.

As far as applying to RGB hosts, the limiting factor for the usefulness of `vespa` at this stage *may* be `isochrones`' performance on evolved stars, which is fairly spotty, given the way it's doing interpolation at this stage (a big area of future improvement; see, e.g. https://github.com/timothydmorton/isochrones/issues/69).  A safety valve for that (that may or may not be necessary to use) if you're getting wacky-looking star fits *might* be actually try giving constraints directly in mass/age/feh, if you happen to have those via another analysis.  Another thing that I'd like to do in `isochrones` before too long is the ability to define some sort of "fixed" stellar models that could allow you to specify exactly the properties of the star instead of having to fit for them.

Thanks for your help so far, and in advance for anything else you are able to contribute.  Stay tuned also for further updates from my end; I may end up getting to some of these things before you'd be able to.

**Comment by timothydmorton** [comment]
@aarondotter I can confirm that `UBVRIPlus` contains `TESS` magnitudes in MIST v1.1 (which is now included in [isochrones](github.com/timothydmorton/isochrones).)

**Comment by campante80** [comment]
@timothydmorton In #26, you've mentioned a development branch focusing on the application of VESPA to TESS photometry. Can't seem to find it. Could you please point me to it? Cheers, Tiago

**Comment by timothydmorton** [comment]
Hi, sorry-- I had neglected to push it to github; it is the 'cadence' branch.  I *believe* it should work with the current master branch of isochrones....

**Comment by priyashmistry** [comment]
@campante80 , @timothydmorton I'm trying VESPA 0.6 on TESS confirmed planets, but don't know the structure of input files. If possible can you please provide an example file similar to kep22 (In example folder).
